### DISCUSSION TITLE
FDTD Simulation for Perovskite Solar Cells

### DISCUSSION THREAD

**Discussion by adlansubejo** [NanoComp/meep#2847] [Q&A] [answered]
Dear All

I'd like to know how to define the medium (material) for simulating perovskite solar cells.

Should I use `E_susceptibilities` like this example, [example eps_fit](https://github.com/NanoComp/meep/blob/master/python/examples/eps_fit_lorentzian.py)?

Please let me know if this approach is good or suggest an alternative method.

Thank you for the advice

**Comment by stevengj** [comment]
You have to do some kind of fit.  Exactly how you do this depends on the bandwidth you are interested in, and what the material dispersion ε(ω) looks like in that bandwidth for your material(s).

**Reply by adlansubejo** [reply]
So, if I have EPS material with a wavelength range of 300-800 nm. Am I must use Lorentz fit so I can get the `E_susceptibilities` properties from the range of EPS?

**Reply by oskooi** [reply, marked as answer]
Check out this [script](https://github.com/NanoComp/meep/blob/master/python/examples/eps_fit_lorentzian.py) which does the fitting via gradient-based optimization using [NLopt](https://nlopt.readthedocs.io/en/latest/) for a complex refractive index $\tilde{n}(\lambda)$ supplied as a CSV file. The script generates a `meep.Medium` object as output.

**Reply by dalarev** [reply]
Is there a way to save this meep.Medium output so that the code need not run every time? Using a large number of Lorentzian terms can take a while to compute.
